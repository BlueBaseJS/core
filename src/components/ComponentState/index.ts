import ComponentState from './ComponentState';
export * from './ComponentState';
export default ComponentState;
