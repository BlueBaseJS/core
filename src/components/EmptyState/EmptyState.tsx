import { BlueBase } from '../../BlueBase';
import { BlueBaseContext } from '../../Context';
import React from 'react';

export class EmptyState extends React.PureComponent {

	static contextType = BlueBaseContext;

	render() {

		// FIXME: remove typecasting, added because current react typings don't seem to support this.context
		const BB: BlueBase = (this as any).context;

		return (
			<BB.Components.ComponentState title="Empty area" description="No data found." />
		);
	}
}
