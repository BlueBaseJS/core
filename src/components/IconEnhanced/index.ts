import IconEnhanced from './IconEnhanced';
export default IconEnhanced;
