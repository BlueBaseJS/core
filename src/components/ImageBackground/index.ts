import ImageBackground from './ImageBackground';
export default ImageBackground;
export * from './ImageBackground';
