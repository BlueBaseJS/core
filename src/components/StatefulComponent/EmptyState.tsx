import { BlueRainType, withBlueRain } from '../../index';
import React from 'react';

const EmptyState = (props: { bluerain: BlueRainType }) => {

	const BR = props.bluerain;

	const ComponentState = BR.Components.get('ComponentState');

	return (
		<ComponentState
			title="Empty area"
			description="No data found."
		/>
	);
};

export default withBlueRain(EmptyState);
