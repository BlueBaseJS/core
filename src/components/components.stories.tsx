import * as React from 'react';
// tslint:disable-next-line:no-implicit-dependencies
import { storiesOf } from '@storybook/react';
import BR from '../index';
import ComponentStateButton from '../components/ComponentState/ComponentStateButton';

const stories = storiesOf('Avatar', module);

stories.add('ComponentStateButton', () => {
	return (
    <ComponentStateButton/>);
});
