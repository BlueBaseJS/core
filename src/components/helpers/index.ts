import react from 'react';
const isHidden = (screenSize, props) => {
	switch (screenSize) {
		case 'sm':
			return !!props.smHidden;
		case 'md':
			return !!props.mdHidden;
		case 'lg':
			return !!props.lgHidden;
		case 'xl':
			return !!props.xlHidden;
		case 'xs':
			return !!props.xsHidden;
		default:
			return false;
	}
};

const toPercent = num => `${num * 100}%`;

const getComponentWidth = (screenSize, props) => {
	switch (screenSize) {
		case 'sm':
			if (props.sm) {
				return toPercent(props.sm / props.rowSize);
			}
			return props.parentWidth;

		case 'xs':
			if (props.xs) {
				return toPercent(props.xs / props.rowSize);
			}
			return props.parentWidth;

		case 'xl':
			if (props.xl) {
				return toPercent(props.xl / props.rowSize);
			}
			return props.parentWidth;

		case 'md':
			if (props.md) {
				return toPercent(props.md / props.rowSize);
			} else if (props.sm) {
				return toPercent(props.sm / props.rowSize);
			}
			return props.parentWidth;

		case 'lg':
			if (props.lg) {
				return toPercent(props.lg / props.rowSize);
			} else if (props.md) {
				return toPercent(props.md / props.rowSize);
			} else if (props.sm) {
				return toPercent(props.sm / props.rowSize);
			}
			return props.parentWidth;

		default:
			return props.parentWidth;
	}
};

const getComponentOffset = (screenSize, props) => {
	switch (screenSize) {
		case 'sm':
			if (props.smOffset) {
				return toPercent(props.smOffset / props.rowSize);
			}
			return 0;

		case 'xs':
			if (props.xsOffset) {
				return toPercent(props.xsOffset / props.rowSize);
			}
			return 0;

		case 'xl':
			if (props.xlOffset) {
				return toPercent(props.xlOffset / props.rowSize);
			}
			return 0;

		case 'md':
			if (props.mdOffset) {
				return toPercent(props.mdOffset / props.rowSize);
			} else if (props.smOffset) {
				return toPercent(props.smOffset / props.rowSize);
			}
			return 0;

		case 'lg':
			if (props.lgOffset) {
				return toPercent(props.lgOffset / props.rowSize);
			} else if (props.mdOffset) {
				return toPercent(props.mdOffset / props.rowSize);
			} else if (props.smOffset) {
				return toPercent(props.smOffset / props.rowSize);
			}
			return 0;
		default:
			return 0;
	}
};

export { isHidden, getComponentWidth, getComponentOffset };
