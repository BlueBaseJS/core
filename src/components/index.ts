export * from './BlueBaseApp';
export * from './BlueBaseContent';
export * from './BlueBaseFilter';
export * from './ComponentState';
export * from './DataObserver';
export * from './DynamicIcon';
export * from './EmptyState';
export * from './ErrorObserver';
export * from './ErrorState';
export * from './Header';
export * from './HeaderBackButton';
export * from './HeaderTitle';
export * from './HomeScreen';
export * from './HoverObserver';
export * from './JsonSchema';
export * from './Link';
export * from './LoadingState';
export * from './Navigation';
export * from './NavigationActions';
export * from './Noop';
export * from './PluginIcon';
export * from './ReactLoadableLoading';
export * from './Redirect';
export * from './StatefulComponent';
export * from './WaitObserver';
