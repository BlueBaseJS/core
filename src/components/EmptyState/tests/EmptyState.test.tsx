import * as Component from '../../../components';
import * as Native from '../../../native';
import { BlueBase } from '../../../BlueBase';
import { BlueBaseProvider } from '../../../Context';
import { EmptyState } from '../EmptyState';
import React from 'react';
import TestRenderer from 'react-test-renderer';

beforeEach(() => {
	jest.resetModules();
});

const BB = new BlueBase();
BB.Components.register('View', Native.View);
BB.Components.register('Text', Native.Text);
BB.Components.register('Image', Native.Image);
BB.Components.register('ComponentState', Component.ComponentState);

describe('EmptyState', () => {
	const EmptyStateWithProvider = (props: any) => (
		<BlueBaseProvider value={BB}>
			<EmptyState {...props}/>
		</BlueBaseProvider>
	);
	test(`Snapshot EmptyState`, () => {
		const component = TestRenderer.create(
			<EmptyStateWithProvider/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

});
