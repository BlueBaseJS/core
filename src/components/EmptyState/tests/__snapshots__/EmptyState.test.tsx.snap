// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Snapshot EmptyState 1`] = `
<View>
  <Text>
    Empty area
  </Text>
  <Text>
    No data found.
  </Text>
</View>
`;
